client-core/src/service/key_service.rs

```diff
@@ -56,7 +92,152 @@ where
 
     /// Clears all storage
     pub fn clear(&self) -> Result<()> {
-        self.storage.clear(KEYSPACE)
+        self.storage.clear(KEYSPACE).expect("clear keyspace");
+        self.storage
+            .clear(KEYSPACE_HD)
+            .expect("clear keyspace for hd");
+        Ok(())
+    }
+
+    /// get random mnemonic
+    pub fn get_random_mnemonic(&self) -> String {
+        let mnemonic = Mnemonic::new(MnemonicType::Words24, Language::English);
+        mnemonic.clone().to_string()
+    }
+
+    /// is hd
+    pub fn is_hd_wallet(&self, name: &str, passphrase: &SecUtf8) -> bool {
```

Comment:
Instead of having `is_hd_wallet`, can we have a function `get_wallet_kind` which returns `WalletKind` enum?

Reply:
ok